### PR TITLE
Add pdm + missing deps: atoml, installer, pdm-pep517,pythonfinder

### PR DESCRIPTION
**Pull request by hadim**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/main/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by hadim** [comment]
This is required to move from with https://github.com/conda-forge/mkdocstrings-feedstock/pull/12#issuecomment-926271336

**Comment by hadim** [comment]
@conda-forge/staged-recipes : `pdm` is a packaging tool similar to `poetry`. I need it for https://github.com/conda-forge/mkdocstrings-feedstock/pull/12#issuecomment-926271336. The issue is that `pdm` depends on `atoml` and `atoml` requires `pdm` for its installation.

**Comment by hadim** [comment]
@conda-forge/staged-recipes LGTM

File: recipes/installer/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+{% set name = "installer" %}
```
Comment (ocefpaf) [inline recipes/installer/meta.yaml:1]: I wonder if we should namespace this one and call it `python-installer` :-/

Comment (hadim) [reply recipes/installer/meta.yaml:1]: No hard opinion on that point so let me know what you prefer or we can wait for more inputs.

Comment (bandersen23) [reply recipes/installer/meta.yaml:1]: @ocefpaf What about `pdm-installer` to be more specific? Not sure what is easiest.

Comment (ocefpaf) [reply recipes/installer/meta.yaml:1]: If that makes sense, sure. My only concern is with "installer" alone. Too generic.

Comment (hadim) [reply recipes/installer/meta.yaml:1]: `installer` is not specific to `pdm` (unless I miss something here) so I don't think it should be namespace with `pdm-`. `python-installer` seems more generic here.

Comment (hadim) [reply recipes/installer/meta.yaml]: I have renamed it to `python-installer`.

**Comment by hadim** [comment]
@ocefpaf LGTM

**Pull request merged**